### ISSUE TITLE
The tool currently does not analyze email headers for common structural anomalies

### ISSUE THREAD

**Issue by keraattin** [keraattin/EmailAnalyzer#71]
## Enhancement Description

The tool currently does not analyze email headers for common structural anomalies that are strong indicators of phishing, spoofing, or bulk spam tooling. Checking for these patterns would give analysts an immediate flag without requiring manual header inspection.

## Expected Behavior or Results

When investigation mode is active, a **Suspicious Headers** entry should be added to the headers investigation when any of the following are detected:

| Pattern | Why it matters |
|---|---|
| Missing `Message-ID` header | Legitimate MTAs always generate one; its absence suggests a script or spoofed email |
| Missing `MIME-Version` header | Expected in all modern emails with non-plain content |
| `Date` header more than 2 days in the future | Timestamp manipulation to evade spam filters |
| `Date` header more than 30 days in the past | May indicate a replayed or manipulated message |
| `X-Mailer` containing known bulk/spam tools | e.g. values like `The Bat!`, `PHPMailer`, mass-mailer strings |

## Example

```
[Suspicious Headers]
Missing Message-ID: True
Date Anomaly: Date is 45 days in the past (2024-11-01)
```

## Desktop
- OS with Version: Any
- Python Version: Python 3.10+
- EmailAnalyzer Project Version: v2.0

## Additional context
Affected function: `get_headers()` in `email-analyzer.py`. All checks can be performed using the already-parsed headers dict and Python's `email.utils.parsedate_to_datetime()` for date parsing. No new dependencies required.
